### PR TITLE
Fix issue with Adding Local DNS Records does not add to /etc/pihole/custom.list when using grep 3.10

### LINKED ISSUE #5268: Adding Local DNS Records does not add to /etc/pihole/custom.list
### Versions

  Pi-hole version is v5.16.2 (Latest: v5.16.2)
  AdminLTE version is v5.19 (Latest: v5.19)
  FTL version is v5.22 (Latest: v5.22)

### Platform

- OS and version:  Manjaro

- Platform:  Raspberry Pi4

### Expected behavior

Adding DNS record adds entry to /etc/pihole/custom.list

### Actual behavior / bug

Record not added to /etc/pihole/custom.list

### Steps to reproduce

1. Go to "admin/dns_records.php" i.e. "Local DNS Records [A/AAAA]" i.e. Local DNS -> DNS Records page
2. Enter "Domain" and "IP Address"
3. Click "Add"
4. Note entry not added to "List of local DNS domains"

## Debug Token 

- URL:  

"Pi-hole debug options are disabled.

ArchLinux distribution of Pi-hole is not supported by official project and this debug report CAN NOT be uploaded to their debug team."

### Additional context

```
# /usr/bin/pihole -a addcustomdns 172.16.0.1 test.telegram.org false
  [✓] Adding custom DNS entry...
  [✗] Invalid Domain passed!
```

Issue may be related to:

[Code](https://github.com/pi-hole/pi-hole/blob/3c86af0e5938e5b03bc12abe9563b8d1efa113d8/advanced/Scripts/webpage.sh#L620)

```
checkDomain()
{
    local domain validDomain
    # Convert to lowercase
    domain="${1,,}"
    validDomain=$(grep -P "^((-|_)*[a-z\\d]((-|_)*[a-z\\d])*(-|_)*)(\\.(-|_)*([a-z\\d]((-|_)*[a-z\\d])*))*$" <<< "${domain}") # Valid chars check
    validDomain=$(grep -P "^[^\\.]{1,63}(\\.[^\\.]{1,63})*$" <<< "${validDomain}") # Length of each label
    echo "${validDomain}"
}
```

### PR DESCRIPTION
**Pull request by MrDuck2742**
Fixes #5268

### CODE REVIEW

**Comment by rdwebdesign** [comment]
Can you please sign off your commit to satisfy the DCO bot requirement? 

- https://docs.pi-hole.net/guides/github/dco/
- https://docs.pi-hole.net/guides/github/how-to-signoff/

**Comment by yubiuser** [comment]
Thanks for your PR. It contains 4 commits, but only the last one contains your changes. It seems, you started from an older `development` base branch. Please update your base branch and rebase your commit.

**Comment by yubiuser** [comment]
`development` is the branch you want.

**Comment by MrDuck2742** [comment]
Sorry about all the changes! Just getting to grips with Git ...

**Comment by MrDuck2742** [comment]
ok ... about ... now :o)

File: advanced/Scripts/webpage.sh
Comment (MrDuck2742) [inline advanced/Scripts/webpage.sh:1]: Substitute //d for 0-9

**Review by yubiuser — APPROVED** [review]

**Comment by MrDuck2742** [comment]
Just signed with GPG key ... should _all_ be good now ...

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**